src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,72 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import './CurrentUser.scss';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+import { getUser } from '../../api';
+
+interface Props {
+  userId: number;
+  onButtonClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const CurrentUser: React.FC<Props> = ({ userId, onButtonClick }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const loadData = async () => {
+    setIsError(false);
+    setIsLoading(true);
+    await getUser(userId)
+      .then(response => setUser(response))
+      .catch(() => setIsError(true));
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    loadData();
+  }, [userId]);
+
+  return (
+    <>
+      {isError
+        ? (<h2>Could not get user info</h2>)
+        : (
+          <div className="CurrentUser">
+            {isLoading
+              ? (<h2>Loading</h2>)
```

Comment:
guess it'll work that way
```suggestion
              ? <h2>Loading</h2>
```